fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/ProtocolMetricDataApiReqestImpl.java

```diff
@@ -0,0 +1,435 @@
+// Copyright 2020 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.apirequest;
+
+import static com.yahoo.bard.webservice.web.ErrorMessageFormat.METRICS_MISSING;
+
+import com.yahoo.bard.webservice.config.BardFeatureFlag;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionDictionary;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.data.metric.LogicalMetric;
+import com.yahoo.bard.webservice.data.metric.MetricDictionary;
+import com.yahoo.bard.webservice.data.time.Granularity;
+import com.yahoo.bard.webservice.data.time.GranularityParser;
+import com.yahoo.bard.webservice.druid.model.builders.DruidFilterBuilder;
+import com.yahoo.bard.webservice.druid.model.orderby.OrderByColumn;
+import com.yahoo.bard.webservice.logging.RequestLog;
+import com.yahoo.bard.webservice.logging.TimedPhase;
+import com.yahoo.bard.webservice.table.LogicalTable;
+import com.yahoo.bard.webservice.table.LogicalTableDictionary;
+import com.yahoo.bard.webservice.web.ApiHaving;
+import com.yahoo.bard.webservice.web.ResponseFormatType;
+import com.yahoo.bard.webservice.web.apirequest.exceptions.BadApiRequestException;
+import com.yahoo.bard.webservice.web.apirequest.generator.having.HavingGenerator;
+import com.yahoo.bard.webservice.web.apirequest.generator.metric.ProtocolLogicalMetricGenerator;
+import com.yahoo.bard.webservice.web.filters.ApiFilters;
+import com.yahoo.bard.webservice.web.util.BardConfigResources;
+import com.yahoo.bard.webservice.web.util.PaginationParameters;
+
+import org.joda.time.DateTimeZone;
+import org.joda.time.Interval;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.ws.rs.core.PathSegment;
+
+/**
+ * ProtocolMetricDataApiRequest supports the parameterized metric contracts used by protocol metrics.
+ */
+public class ProtocolMetricDataApiReqestImpl extends DataApiRequestImpl {
```

Comment:
1- This is a subclass of DARI, people can opt into it intentionally.  Otherwise they'll have requests that are processed in a legacy compatible way.  We can deprecate the other processing style at leisure.
2- Ideally we want people to migrate onto DARFactories and those are much more in the control of the customers.

Reply:
I thought we wanted all clients to get the protocol metric updates. Wasn't that the point of changing the original logical metric parsing method from static to non-static?

Reply:
Right now there's no good way to inject the Metric building code into the base ApiRequestImpl.  Thoughts?